includes/class-freemius.php

```diff
@@ -4739,24 +4747,42 @@ private function parse_settings( &$plugin_info ) {
                 $this->_plugin :
                 new FS_Plugin();
 
+            $premium_slug   = $this->get_option( $plugin_info, 'premium_slug', "{$this->_slug}-premium" );
+            $premium_suffix = $this->get_option( $plugin_info, 'premium_suffix', '(Premium)' );
+
             $plugin->update( array(
                 'id'                   => $id,
                 'type'                 => $this->get_option( $plugin_info, 'type', $this->_module_type ),
                 'public_key'           => $public_key,
                 'slug'                 => $this->_slug,
+                'premium_slug'         => $premium_slug,
                 'parent_plugin_id'     => $parent_id,
                 'version'              => $this->get_plugin_version(),
-                'title'                => $this->get_plugin_name(),
+                'title'                => $this->get_plugin_name( $premium_suffix ),
                 'file'                 => $this->_plugin_basename,
                 'is_premium'           => $this->get_bool_option( $plugin_info, 'is_premium', true ),
+                'premium_suffix'       => $premium_suffix,
                 'is_live'              => $this->get_bool_option( $plugin_info, 'is_live', true ),
                 'affiliate_moderation' => $this->get_option( $plugin_info, 'has_affiliation' ),
             ) );
 
             if ( $plugin->is_updated() ) {
+                $register_activation_hook = ( ! is_object( $this->_plugin ) || $this->_plugin->premium_slug !== $premium_slug );

```

Comment:
@fajardoleo: 
1. I don't understand this addition, can you please explain it?
2. Also, the name of the method is `parse_settings()` so the registration of the activation hook should not be here anyway.

Reply:
@vovafeldman, when the settings are parsed and the premium slug doesn't match the stored one, the activation hook needs to be registered again since the previous registration logic doesn't have the correct premium version's directory name anymore (`_register_hooks` is called directly in the constructor and it registers an activation hook before the settings are parsed).